src/command/dev/protocol/leader.rs

```diff
@@ -374,9 +393,10 @@ impl LeaderSession {
         socket_write(&message, stream)
     }
 
-    /// Gets the supergraph configuration from the internal state.
-    /// Calling `.to_string()` on a [`SupergraphConfig`] writes
-    fn supergraph_config(&self) -> SupergraphConfig {
+    /// Gets the supergraph configuration from the internal state. This can different from the
+    /// supergraph.yaml file as it represents intermediate states of composition while adding
+    /// subgraphs to the internal representation of that file
+    fn supergraph_config_internal_representation(&self) -> SupergraphConfig {
```

Comment:
renamed this private method because I found it super confusing (it's not the supergraph config in terms of the yaml, but rather the unfolding supergraph config as it's being built out of subgraphs added to it--ie, it can represent intermediate states of the yaml version of the config); updated the comment, too, because there's no longer a `to_string()`